tests/models/test_modeling_voxtral.py

```diff
@@ -0,0 +1,181 @@
+        """
+        model_id = "mistralai/Voxtral-Mini-3B-2507"
+        config = AutoConfig.from_pretrained(model_id)
+        module = load_multimodal_text_to_text_model(
+            model_id,
+            use_custom_sdpa=True,
+            use_custom_kv_cache=True,
+            qlinear="8da4w",
+            qembedding="8w",
+        )
+
+        ep = module.export()
+
+        # Generate
+        tokenizer = AutoTokenizer.from_pretrained(model_id)
+        conversation = [
+            {
+                "role": "user",
+                "content": [
+                    {
+                        "type": "audio",
+                        "url": "https://huggingface.co/datasets/eustlb/audio-samples/resolve/main/dude_where_is_my_car.wav",
+                    },
+                    {"type": "text", "text": "What can you tell me about this audio?"},
+                ],
+            }
+        ]
+        processor = AutoProcessor.from_pretrained(model_id)
+        inputs = processor.apply_chat_template(conversation)
+
+        input_ids = inputs["input_ids"]
+        token_embeddings = ep["token_embeddings"].module().forward(input=input_ids)
+
+        if "input_features" in inputs:
+            audio_embeddings = (
+                ep["audio_encoder"]
+                .module()
+                .forward(
+                    input_features=inputs["input_features"],
+                    # inputs_embeds=token_embeddings,
+                    # input_ids=inputs["input_ids"],
+                )
+            )
+
+        audio_token_mask = inputs["input_ids"] == config.audio_token_id
+        token_embeddings[audio_token_mask] = audio_embeddings
+
+        # Prefill prompt embeddings
+        logits = (
+            ep["decoder"]
+            .module()
+            .forward(
+                inputs_embeds=token_embeddings,
+                cache_position=torch.arange(token_embeddings.shape[1], dtype=torch.long),
+            )
+        )
+
+        token = torch.argmax(logits[:, -1, :])
+
+        tokens = [token.item()]
+        print(tokenizer.decode([token.item()]), end="")
+
+        pos = token_embeddings.shape[1]
+
+        max_generation_len = 64
+        while pos < input_ids.shape[-1] + max_generation_len:
+            token_embedding = ep["token_embeddings"].module().forward(input=token.unsqueeze(0).unsqueeze(0))
+            logits = (
+                ep["decoder"]
+                .module()
+                .forward(
+                    inputs_embeds=token_embedding,
+                    cache_position=torch.tensor([pos], dtype=torch.long),
+                )
+            )
+            token = torch.argmax(logits[:, -1, :])
+            print(tokenizer.decode([token.item()]), end="")
+            tokens.append(token.item())
+            pos += 1
```

Comment:
we should use the generate method from modeling.py class?

Reply:
This is just for ExportedProgram

Reply:
I understand but can you not do generate on the wrapped module that wraps exportedPrograms?